File: ArrayJAVA.cpp
#include<iostream>
using namespace std;

class ArrayX
{
	private:
			int *Arr;
			int iSize;
	public: 
			ArrayX(int iValue)		//Parameterized Constructor
			{
				this->iSize = iValue;
				Arr = new int[iSize];	//Resource Allocation
			}
			
			~ArrayX()		//Destructor
			{
				delete[]Arr;	//Deallocate the resources
			}
			void Accept()
			{
				int iCnt=0;
				cout<<"Enter The Elements:"<<endl;
				
				for(iCnt=0;iCnt<iSize;iCnt++)
				{
					cin>>Arr[iCnt];
				}
			}
			void Display()
			{
				int iCnt=0;
				for(iCnt=0;iCnt<iSize;iCnt++)
				{
					cout<<Arr[iCnt]<<"\t"<<endl;
				}
			}
			
			int Sumation()
			{
				int iCnt=0;
				int iSum=0;
				for(iCnt=0;iCnt<iSize;iCnt++)
				{
					iSum = iSum + Arr[iCnt];
				}		
				return iSum;
			}
};

int main()
{
	auto int iRet =0 ;
	ArrayX obj(5);
	obj.Accept();
	obj.Display();	//Display(&obj)
	
	iRet = obj.Sumation();
	cout<<"Sumation is "<<iRet<<endl;
	return 0;
}

File: ArrayJAVA.java
import java.lang.*;
import java.util.*;


class ArrayJava
{
	public static void main(String a[])
	{
		int iRet=0;
		int iLength=0;
		
		Scanner sobj = new Scanner(System.in);
		
		System.out.println("Enter the Value for Constructor:");
		iLength = sobj.nextInt();
		ArrayX obj = new ArrayX(iLength);
		
		obj.Accept();
		obj.Display();
		
		iRet= obj.Sumation();
		System.out.println("Sum of elements: "+iRet);
		
		obj = null;
	}
}

class ArrayX
{
	private int Arr[];
	
	public ArrayX(int iNo)
	{
		Arr = new int[iNo];
	}
	
	public void Accept()
	{
		int iCnt=0;
		Scanner sobj = new Scanner(System.in);
		System.out.println("Enter the values: ");
		
		for(iCnt=0;iCnt<Arr.length;iCnt++)
		{
			Arr[iCnt] = sobj.nextInt();
		}
	}
	
	public void Display()
	{
		int iCnt=0;
		System.out.println("Values are: ");
		
		for(iCnt=0;iCnt<Arr.length;iCnt++)
		{
			System.out.println(Arr[iCnt]);
		}
	}
	
	public int Sumation()
	{
		int iCnt=0, iSum=0;
		
		for(iCnt=0;iCnt<Arr.length;iCnt++)
		{
		iSum= iSum+Arr[iCnt];
		}
		return iSum;
	}
	
}